lib/engine-addon.js

```diff
@@ -738,27 +738,29 @@ module.exports = {
           otherAssets = this.otherAssets();
         }
 
-        // Merge all of our final trees!
-        var finalMergeTrees = [
-          publicRelocated,
-          addonsEnginesPublicTreesMerged,
-          otherAssets,
-          finalStylesTree,
-          vendorJSImportTree,
-          concatEngineTree
-        ];
-
-        // Separate engines routes from the host if includeRoutesInApplication
-        // is false
-        var separateRoutes = this.lazyLoading.includeRoutesInApplication === false;
-        if (separateRoutes) {
```

Comment:
I believe the tests are failing partially due to the collisions here. Basically, all the old stuff should still be in the addon we just want to add `configEnvironment` from the new part to the `finalMergeTrees`, does that make sense?